include_core/omrsig.h

```diff
@@ -32,17 +32,12 @@
 extern "C" {
 #endif /* __cplusplus */
 
-#if defined(OSX)
-#define __THROW
-#endif /* defined(OSX) */
```

Comment:
Replacing `omrsig.cpp` with `omrsig.c` seems to be a bigger task than I expected.

This should never have had made reference to `__THROW` (symbols starting with two underscores are the domain of the compiler, not application code). Instead, we should define a new symbol which it an empty throws clause for C++ and nothing for C: see #7807.

Reply:
Looks promising to try,
Simply having `#define __THROW` with the Open XL compiler was enough for it to complain earlier. Hoping that the new symbol only definining the `throw()` specifier for `cplusplus` context will be enough.

Will have to see with XLC build too though, as soon as I removed the define entirely the last time, the XLC vm builds broke too on jenkins.